lib/syslogger/creators.rb

```diff
@@ -4,15 +4,15 @@
 module SysLogger
   module Creators
     def self.unix_dgram_socket(socket_path)
-      proc {
+      Proc.new {
```

Comment:
Don't get me wrong -- I think the entire logger interface in Ruby is braindead and awful. I would much prefer that we drop the arguments to `initialize` entirely and switch to something sane. It's a balancing act between having an interface which is vaguely similar to the existing (again, imo absolutely awful) interface and doing the right thing.